apis/src/pages/tournament_create.rs

```diff
@@ -249,17 +251,35 @@ pub fn TournamentCreate() -> impl IntoView {
                         />
                     </div>
 
-                    <div class="flex flex-col">
-                        Description:
-                        <textarea
-                            class="px-3 py-2 w-10/12 leading-tight rounded border shadow appearance-none focus:outline-none"
-                            name="Tournament description"
-                            type="text"
-                            prop:value=tournament.description
-                            placeholder="At least a 50 character description"
-                            on:input=update_from_input(tournament.description)
-                            attr:maxlength="2000"
-                        ></textarea>
+                    <div class="flex flex-col mt-4">
+                        <div class="flex flex-row">
+                            <button
+                                on:click=move |_| is_not_preview_desc.update(|b| *b = !*b)
+                                class="flex gap-1 justify-center items-center px-4 font-bold text-white rounded bg-button-dawn dark:bg-button-twilight hover:bg-pillbug-teal active:scale-95 disabled:opacity-25 disabled:cursor-not-allowed disabled:hover:bg-transparent"
+                            >
+                                {move || if is_not_preview_desc() { "Preview" } else { "Edit" }}
+                            </button>
+                            <span class="px-1 font-bold">Description:</span>
+
+                        </div>
+                        <Show
+                            when=is_not_preview_desc
+                            fallback=move || {
+                                view! {
+                                    <div class="prose dark:prose-invert" inner_html=markdown_desc />
```

Comment:
This needs some extra classes otherwise when you switch to preview the generated html can flow out of it's parent div and push stuff around in weird ways. Something like this seems to work well:

```html
<div class="p-4 w-full break-words prose dark:prose-invert" inner_html=markdown_desc />
```